lib/widgets/message_list.dart

```diff
@@ -345,6 +351,67 @@ class ScrollToBottomButton extends StatelessWidget {
   }
 }
 
+class MarkAsReadWidget extends StatelessWidget {
+  const MarkAsReadWidget({super.key, required this.model});
+
+  final MessageListView model;
```

Comment:
Hmm, does this really want a `MessageListView`? I'd generally like to let that remain in the hands of the `_MessageListState` — that's the element for which the `MessageListView` is the view-model.

It looks like this widget only consults `model.narrow` and `model.store`. So let's instead just pass it the narrow, and it can get the store in the usual way.